test/routing/VersionRoutingRequestHandlerSpec.scala

```diff
@@ -17,36 +17,69 @@
 package routing
 
 import com.typesafe.config.ConfigFactory
-import mocks.MockAppConfig
-import org.scalamock.handlers.CallHandler0
-import org.scalatest.Inside
+import config.AppConfig
+import org.mockito.ArgumentMatchers
+import org.mockito.Mockito.{reset, when}
+import org.scalatest.{BeforeAndAfterEach, Inside}
+import org.scalatestplus.mockito.MockitoSugar.mock
 import play.api.Configuration
 import play.api.http.HeaderNames.ACCEPT
 import play.api.http.{HttpConfiguration, HttpFilters}
 import play.api.libs.json.Json
-import play.api.mvc.{EssentialAction, _}
+import play.api.mvc._
 import play.api.routing.Router
-import play.api.routing.Router.Routes
 import play.api.test.FakeRequest
 import play.api.test.Helpers._
 import uk.gov.hmrc.play.audit.http.connector.AuditConnector
 import uk.gov.hmrc.play.bootstrap.config.HttpAuditEvent
 import utils.{BaseSpec, ErrorHandler}
 import v1.models.errors.ErrorResponses._
 
-class VersionRoutingRequestHandlerSpec extends BaseSpec with Inside with MockAppConfig {
-  test =>
+class VersionRoutingRequestHandlerSpec extends BaseSpec with Inside with BeforeAndAfterEach {
 
-  private val defaultRouter: Router = mock[Router]
-  private val v1Router: Router      = mock[Router]
-  private val v2Router: Router      = mock[Router]
-  private val v3Router: Router      = mock[Router]
+  import play.api.mvc.Handler
+  import play.api.routing.sird._
 
-  private val routingMap = new VersionRoutingMap {
-    override val defaultRouter: Router    = test.defaultRouter
+  object DefaultHandler extends Handler
+  object V1Handler extends Handler
+  object V2Handler extends Handler
+  object V3Handler extends Handler
+
+  private val defaultRouter = Router.from { case GET(p"") =>
+    DefaultHandler
+  }
+
+  private val v1Router = Router.from { case GET(p"/v1") =>
+    V1Handler
+  }
+
+  private val v2Router = Router.from { case GET(p"/v2") =>
+    V2Handler
+  }
+
+  private val v3Router = Router.from { case GET(p"/v3") =>
+    V3Handler
+  }
+
+  private def routingMap(router: Router) = new VersionRoutingMap {
+    override val defaultRouter: Router    = router
     override val map: Map[String, Router] = Map("1.0" -> v1Router, "2.0" -> v2Router, "3.0" -> v3Router)
   }
 
+  private val mockAppConfig: AppConfig = mock[AppConfig]
+
+  override def beforeEach(): Unit = {
+    reset(mockAppConfig)
+
+    when(mockAppConfig.apiGatewayContext).thenReturn("gateway")
+    when(mockAppConfig.featureSwitch).thenReturn(Some(Configuration(ConfigFactory.parseString("""
+                                                                                                |version-1.enabled = true
+                                                                                                |version-2.enabled = true
+      """.stripMargin))))
+    when(mockAppConfig.apiStatus(ArgumentMatchers.eq("1.0"))).thenReturn("ALPHA")
+    when(mockAppConfig.endpointsEnabled).thenReturn(true)
+  }
```

Comment:
Same here. Please move the mocking on lines 74 to 80 under the `class Test(implicit acceptHeader: Option[String])` and delete the rest, thanks.